backend/pkg/api/connect/service/monitoring/v1/service.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package monitoring contains all handlers for the monitoring endpoints
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+	"time"
+
+	adminv2 "buf.build/gen/go/redpandadata/core/protocolbuffers/go/redpanda/core/admin/v2"
+	"connectrpc.com/connect"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	apierrors "github.com/redpanda-data/console/backend/pkg/api/connect/errors"
+	"github.com/redpanda-data/console/backend/pkg/config"
+	redpandafactory "github.com/redpanda-data/console/backend/pkg/factory/redpanda"
+	v1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1"
+	"github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1/dataplanev1connect"
+)
+
+const defaultLimit = uint32(30)
+
+var _ dataplanev1connect.MonitoringServiceHandler = (*Service)(nil)
+
+// Service implements MonitoringServiceHandler
+type Service struct {
+	cfg                   *config.Config
+	logger                *slog.Logger
+	redpandaClientFactory redpandafactory.ClientFactory
+}
+
+// NewService instantiates a new Service
+func NewService(cfg *config.Config, logger *slog.Logger, redpandaClientFactory redpandafactory.ClientFactory) *Service {
+	return &Service{cfg: cfg, logger: logger, redpandaClientFactory: redpandaClientFactory}
+}
+
+func kafkaAPIFromKey(key int32) v1.KafkaAPI {
+	val, ok := map[int32]v1.KafkaAPI{
```

Comment:
NIT: Should probably move to package level to avoid the allocation of the entire map for every call.